activerecord/test/cases/associations_test.rb

```diff
@@ -164,6 +164,15 @@ def test_has_many_association_with_composite_foreign_key_loads_records
     assert_includes(comments, sharded_comments(:great_comment_blog_post_one))
   end
 
+  def test_has_many_association_from_a_non_qc_model_loads_records
```

Comment:
Nit: Let's use "query_constraints" instead of "qc". Especially since it's a test, it's probably beneficial to be as clear about intent as possible, acronyms some times introduce friction.
```suggestion
  def test_has_many_association_from_a_non_query_constraints_model_loads_records
```

Reply:
Good point, I'll need to completely change the name since we are not using a "non query constraints model" anymore and using a model with query constraints different from association's query constraints 👍